src/MetaPaymaster.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+
+/* solhint-disable reason-string */
+
+import "@account-abstraction/interfaces/IEntryPoint.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+contract MetaPaymaster is OwnableUpgradeable {
+    IEntryPoint public entryPoint;
+    mapping(address => uint256) public balances;
+    uint256 public total;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(address owner, IEntryPoint _entryPoint) public initializer {
+        __Ownable_init(owner);
+        entryPoint = _entryPoint;
+    }
+
+    function fund(uint256 actualGasCost) external returns (bool) {
+        if (address(this).balance < actualGasCost) {
```

Comment:
I think this function should just revert if it can't fund? why return a bool? if revert, you'll get it for free when the transfer fails

Reply:
my thinking was that we don't necessarily want to fail user operations if the metapaymaster cannot fund... do you think this is better handled by expecting the caller to catch the revert?

Reply:
I would need to see it in action to get a better opinion. Downsides both ways 
1. Users generally expect things to revert if they fail, so could be a footgun that the tx passes but not paid for
2. on the other hand, they might be annoyed we revert and they have to try/catch. Maybe we could talk to a team

Reply:
switched to reverting 👍 and added a `willFund` read method